Library/Homebrew/cmd/list.rb

```diff
@@ -132,7 +132,9 @@ def list
       system_command! "find", args:         args.named.to_default_kegs.map(&:to_s) + %w[-not -type d -print],
                               print_stdout: true
     else
-      args.named.to_default_kegs.each { |keg| PrettyListing.new keg }
+      formula_names, cask_names = args.named.to_resolved_formulae_to_casks
```

Comment:
I think we need to keep the `to_default_kegs` functionality because it works differently than a "plain" `to_formulae` which is what's used in `to_resolved_formulae_to_casks`

```suggestion
      formula_names, cask_names = args.named.to_resolved_formulae_to_casks(method: :default_kegs)
```

CC @cnnrmnn for a second opinion on this

Reply:
`to_resolved_formulae_to_casks` doesn't take a `method` argument because it passes `method: :resolve` to `to_formulae_to_casks` no matter what. This returns formula instance(s). This is an issue because `PrettyListing` expects a keg. I suspect that `brew list <formula-name>` currently errors out. `PrettyListing` prints out directories and files in specific directories you'd expect to find in a keg (see [here](https://github.com/Homebrew/brew/blob/a9b275d7574a65299a41bf5dabfa3c76736b7074/Library/Homebrew/cmd/list.rb#L182)).

I may be missing some method that would be helpful here, but you might just need to make two calls to resolve casks and formula kegs. Alternatively, you could write some new method or modify an existing method.

Reply:
Ah, good point. How about this?

```suggestion
      formula_names, cask_names = args.named.to_formulae_to_casks(method: :default_kegs)
```

Reply:
Actually that should work.